src/tequila/simulators/simulator_api.py

```diff
@@ -176,9 +176,9 @@ def pick_backend(backend: str = None, samples: int = None, noise: NoiseModel = N
                 for f in INSTALLED_SAMPLERS.keys():
                     return f
         else:
-            if samples is None:
```

Comment:
Will lead to too many clashes with other applications.
return type of tq.simulate(circuit) is not clear from context anymore (wavefunction or density matrix)

Propose the following: Add simulate_density_matrix method. Within the method, check backend and only allow qiskit (because the simulation is not supported for others currently, as far as I understand).